File: 1-1000/0822-card-flipping-game/0822-card-flipping-game.cpp
class Solution {
public:
    int flipgame(vector<int>& fronts, vector<int>& backs) {
        unordered_set<int> bad;
        for(int i=0;i<fronts.size();i++) if(fronts[i]==backs[i]) bad.insert(fronts[i]);
        int ans=INT_MAX;
        for(int x: fronts) if(!bad.count(x)) ans=min(ans,x);
        for(int x: backs) if(!bad.count(x)) ans=min(ans,x);
        return ans==INT_MAX?0:ans;
    }
};

File: 1-1000/0822-card-flipping-game/0822-card-flipping-game.java
class Solution {
    public int flipgame(int[] f, int[] b) {
    boolean con[] = new boolean[2001];
    int n = f.length;
    for(int i=0; i<n; i++){
     if(f[i]==b[i]) con[f[i]] = true;
    }
    int ans = Integer.MAX_VALUE;
    for(int i=0; i<n; i++){
    if(!con[f[i]]) ans = Math.min(ans, f[i]);
    if(!con[b[i]]) ans = Math.min(ans, b[i]);
    }
    return ans==Integer.MAX_VALUE?0:ans;
    }
}